plugins/android-boot/fu-android-boot-device.c

```diff
@@ -0,0 +1,372 @@
+/*
+ * Copyright (C) 2022 Dylan Van Assche <me@dylanvanassche.be>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include <fcntl.h>
+#include <string.h>
+
+#include "fu-android-boot-device.h"
+
+#define ANDROID_BOOT_BOOT_SLOT_ARG    "androidboot.slot_suffix"
+#define ANDROID_BOOT_SERIAL_ARG	      "androidboot.serialno"
+#define ANDROID_BOOT_UNKNOWN_VERSION  "0.0.0"
+#define ANDROID_BOOT_SECTOR_SIZE      512
```

Comment:
maybe just put these values where they're used? They only seem to be used once afterall.

Reply:
I like defines to make them self explanatory, otherwise I feel the need to write a comment for them why these values are used. I rather keep them like this.

Reply:
I think `androidboot.serialno` is way more descriptive and `grep`-able than `ANDROID_BOOT_SERIAL_ARG`

Reply:
But keeping the SECTOR and UNKNOWN_VERSION ones then?

Reply:
yup, that's a fine compromise :)

Reply:
Ohh, you might also want to set the version format -- talking about version strings.

Reply:
Should be already be the case: `fu_device_set_version_format(FU_DEVICE(self), FWUPD_VERSION_FORMAT_PLAIN);` :)